pilot/pkg/serviceregistry/kube/controller/util.go

```diff
@@ -189,3 +190,13 @@ func isNodePortGatewayService(svc *v1.Service) bool {
 	_, ok := svc.Annotations[kube.NodeSelectorAnnotation]
 	return ok && svc.Spec.Type == v1.ServiceTypeNodePort
 }
+
+// Get the proxy pod key which can be used to get pod from the informer cache
+func proxyPodKey(proxy *model.Proxy) string {
+	parts := strings.Split(proxy.ID, ".")
+	if len(parts) == 2 && proxy.Metadata.Namespace != parts[1] {
```

Comment:
what is `proxy.Metadata.Namespace != parts[1]` for?

Reply:
This is to exclude some bad ID (`a.b`) maybe from some tools

Reply:
shouldn't it be `==` not `!=`?

Reply:
And can we have a test covering this case?

Reply:
Sorry i missed the above comment 11days ago.